### PR TITLE
Windows arm32 - fix failing tests or skip

### PR DESCRIPTION
**Pull request by paulmon**

### CODE REVIEW

File: Lib/distutils/_msvccompiler.py [hunk 1]
```diff
@@ -177,6 +177,7 @@ def _find_exe(exe, paths=None):
 PLAT_TO_VCVARS = {
     'win32' : 'x86',
     'win-amd64' : 'x86_amd64',
+    'win-arm' : 'x86_arm',
```
Comment (jslobodzian) [inline Lib/distutils/_msvccompiler.py]: Just double checking x86_arm is the right thing there.  Based on comment above, it seems right..but..

**Comment by jslobodzian** [comment]
>     SKIP_MESSAGE = "These tests are only for win32"

Consider mentioning that these tests do not apply to iot either.

---
Refers to: Lib/distutils/tests/test_msvc9compiler.py:101 in 14d525e. [](commit_id = 14d525ec14723bd2a775c29aa5c3e6c5940983ad, deletion_comment = False)

**Comment by jslobodzian** [comment]
> def zoom_height(top):

Does this actually run on IoT?

---
Refers to: Lib/idlelib/zoomheight.py:20 in 14d525e. [](commit_id = 14d525ec14723bd2a775c29aa5c3e6c5940983ad, deletion_comment = False)

File: PC/pyconfig.h [hunk 2]
```diff
@@ -312,7 +312,11 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
    should define this. */
 #       define HAVE_LARGEFILE_SUPPORT
 #elif defined(MS_WIN32)
+# ifdef _M_ARM
+#       define PLATFORM "win-arm"
+# else // _M_ARM
 #       define PLATFORM "win32"
+# endif // _M_ARM
```
Comment (jslobodzian) [inline PC/pyconfig.h]: Nit: Remove space between # and ifdef/else/endif

File: Lib/test/test_json/test_recursion.py [hunk 3]
```diff
@@ -85,6 +87,7 @@ def test_highly_nested_objects_encoding(self):
         with self.assertRaises(RecursionError):
             self.dumps(d)
 
+    @unittest.skipIf(platform.win32_editionId() == 'NanoServer', "stack overflow on nanoserver stops tests")
```
Comment (jslobodzian) [inline Lib/test/test_json/test_recursion.py]: Do we have a bug for this?  Seems like we should understand why the behavior is different.

File: Lib/test/test_import/__init__.py [hunk 4]
```diff
@@ -641,7 +641,7 @@ def test_trailing_slash(self):
         unload("test_trailing_slash")
 
     # Regression test for http://bugs.python.org/issue3677.
-    @unittest.skipUnless(sys.platform == 'win32', 'Windows-specific')
+    @unittest.skipUnless(sys.platform.startswith('win') and not platform.win32_editionId() == 'NanoServer', 'Windows-specific')
     def test_UNC_path(self):
```
Comment (jslobodzian) [inline Lib/test/test_import/__init__.py]: >def test_UNC_path(self): [](start = 4, length = 24)

UNC paths don't work on Nanoserver?

File: Lib/test/test_platform.py [hunk 5]
```diff
@@ -14,10 +14,11 @@ def test_architecture(self):
         res = platform.architecture()
 
     @support.skip_unless_symlink
+    @unittest.skipIf(platform.win32_is_iot(), "symbolic links don't work on Windows IoT Core or nanoserver")
```
Comment (jslobodzian) [inline Lib/test/test_platform.py]: >symbolic links  [](start = 47, length = 15)

They don't?

File: Lib/distutils/_msvccompiler.py (hunk 1 above)
Comment (paulmon) [reply Lib/distutils/_msvccompiler.py]: yes.  x86 is the build machine architecture and arm is the target architecture for the cross-compiler.

---
In reply to: [224304858](https://github.com/ms-iot/cpython/pull/12#discussion_r224304858) [](ancestors = 224304858)

**Comment by paulmon** [comment]
> def zoom_height(top):

Looks like the answer is no since no work has been done for tkinter.  If there is an enterprise build for arm32 this would still apply.  I might be theoretical but I still think it's the right code.

---
In reply to: [428809186](https://github.com/ms-iot/cpython/pull/12#issuecomment-428809186) [](ancestors = 428809186)

---
Refers to: Lib/idlelib/zoomheight.py:20 in 14d525e. [](commit_id = 14d525ec14723bd2a775c29aa5c3e6c5940983ad, deletion_comment = False)

File: Lib/test/test_json/test_recursion.py (hunk 3 above)
Comment (paulmon) [reply Lib/test/test_json/test_recursion.py]: Created 19271043

---
In reply to: [224306807](https://github.com/ms-iot/cpython/pull/12#discussion_r224306807) [](ancestors = 224306807)

File: Lib/test/test_platform.py (hunk 5 above)
Comment (paulmon) [reply Lib/test/test_platform.py]: The dir command said [SYMBOLIC LINK] but I couldn't run dummy.exe which was a symbolic link to python.exe.  Python was only used to create the symbolic link

---
In reply to: [224308479](https://github.com/ms-iot/cpython/pull/12#discussion_r224308479) [](ancestors = 224308479)

File: Lib/test/test_import/__init__.py (hunk 4 above)
Comment (paulmon) [reply Lib/test/test_import/__init__.py]: More that ACLs are not set to allow it.
C:\>dir \\paulmon-hum\c$\python
Access is denied.

---
In reply to: [224307018](https://github.com/ms-iot/cpython/pull/12#discussion_r224307018) [](ancestors = 224307018)

File: Lib/test/test_json/test_recursion.py (hunk 3 above)
Comment (paulmon) [reply Lib/test/test_json/test_recursion.py]: stack overflows seem fixed.
I took the skips for stack overflows out and the tests all succeeded.
Running full test pass now to verify

---
In reply to: [224525592](https://github.com/ms-iot/cpython/pull/12#discussion_r224525592) [](ancestors = 224525592,224306807)

File: Lib/test/test_json/test_recursion.py [hunk 6]
```diff
@@ -1,5 +1,6 @@
 from test.test_json import PyTest, CTest
-
+import platform
+import unittest
```
Comment (paulmon) [inline Lib/test/test_json/test_recursion.py]: undo changes in this file

**Pull request merged**